### ISSUE TITLE
no-map-without-usage rule does not consider arrow function return shorthand as a return/assignment

### ISSUE THREAD

**Issue by Glavin001** [Glavin001/tslint-clean-code#35]
Code:

```
this.getFiles()
  .then(files => files.map(file => path.join(this.repoPath, file)))
```

Shows incorrect linting error on `files.map`:

```
[tslint] Return value from Array.map should be assigned to a variable. Consider using Array.forEach instead. (no-map-without-usage)
```
